pkg/controller/identity/iamgroupusermembership/controller.go

```diff
@@ -117,11 +118,17 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 		AttachedGroupARN: aws.StringValue(attachedGroupObject.Arn),
 	}
 
+	// This resource is interesting in that it's a binding without its own
+	// external identity. We therefore derive an external name from the
+	// names of the group and user that are bound.
+	meta.SetExternalName(cr, cr.Spec.ForProvider.GroupName+"/"+cr.Spec.ForProvider.UserName)
+
 	cr.SetConditions(xpv1.Available())
 
 	return managed.ExternalObservation{
-		ResourceExists:   true,
-		ResourceUpToDate: true,
+		ResourceExists:          true,
+		ResourceLateInitialized: true,
```

Comment:
Even if it's no-op, the spec update call will override status from what I can tell. We usually handle this operation either in the [initializer itself](https://github.com/crossplane/crossplane-runtime/blob/406fe0b/pkg/reconciler/managed/api.go#L56) or in `Create` by returning [`ExternalNameAssigned: true`](https://github.com/crossplane/crossplane-runtime/blob/f2418dc/pkg/reconciler/managed/reconciler.go#L722).

Given that we'd like to cut a release urgently with these fixes, I'll open another PR with these commits and add a commit changing this to assign in `Create`.

Reply:
I forgot about ExternalNameAssigned! I may be able to tackle that during a meeting today haha.

Reply:
Ugh I was planning to cut the release before you wake up. Anyway I had opened the PR already here with some other small fixes https://github.com/crossplane/provider-aws/pull/728 . Manually tested all affected resources.